### DISCUSSION TITLE
Have multiple flows in file base flow storage

### DISCUSSION THREAD

**Discussion by Marlin-Na** [PrefectHQ/prefect#3033] [Ideas]
Currently `GitHub.get_flow` can only return the first flow created in that script. If you have multiple flows in the script, it may mismatch with the registered flow. I think `flow_location` passed to `get_flow` should not be just python file path, but maybe a tuple of script path and flow name, so that multiple flows within the script can be handled correctly.

**Comment by joshmeek** [comment]
@Marlin-Na thanks for pointing that out, adding the option to provide the flow name will resolve this! Q: what is your use case for having multiple flows in the same file as opposed to having multiple single-flow files?

**Reply by Marlin-Na** [reply]
I think currently `GitHub.get_flow` requires the python script to be self-contained, which means you can't have relative-import or have dependency on any other file in the repo. If I want to reuse certain task/function, I can only copy them to all of the scripts. It makes easier to reuse tasks if multiple flows can be included in one file.

**Reply by joshmeek** [reply]
I see, #2857 is an issue in a similar vein that we are using to track this if you want to put your thoughts on there! We haven't fully solved the idea of being able to use relative imports with this outside of running your flow in a container that has the imports on the PATH. (e.g. https://docs.prefect.io/orchestration/execution/storage_options.html#non-docker-storage-for-containerized-environments)
